tests/unit/test_tap_kafka.py

```diff
@@ -868,6 +870,74 @@ def test_get_timestamp_from_timestamp_tuple(self):
         with pytest.raises(InvalidTimestampException):
             sync.get_timestamp_from_timestamp_tuple("not_a_tuple_or_list")
 
+    def test_initial_start_time_to_offset_reset(self):
+        """Initial start time can be one of 'latest', 'earliest' or an ISO timestamp"""
+        # Earliest should return earliest
+        assert sync.initial_start_time_to_offset_reset('earliest') == 'earliest'
+
+        # Anything else should return latest. Every string, ISO timestamps or non string values
+        assert sync.initial_start_time_to_offset_reset('latest') == 'latest'
+        assert sync.initial_start_time_to_offset_reset('2021-11-01 16:00:30') == 'latest'
+        assert sync.initial_start_time_to_offset_reset(None) == 'latest'
+        assert sync.initial_start_time_to_offset_reset(1234) == 'latest'
+
+    def test_iso_timestamp_to_epoch(self):
+        """Validate converting ISO timestamps to epoch milliseconds"""
+        # Using space as date and time delimiter
+        assert sync.iso_timestamp_to_epoch('2021-11-01 23:01:11') == 1635807671000
+        assert sync.iso_timestamp_to_epoch('2021-11-01 23:01:11.123') == 1635807671123
+        assert sync.iso_timestamp_to_epoch('2021-11-01 23:01:11.123456') == 1635807671123
+        assert sync.iso_timestamp_to_epoch('2021-11-01 23:01:11.123987') == 1635807671123
+        assert sync.iso_timestamp_to_epoch('2021-11-01 23:01:11.123987+00:00') == 1635807671123
+        assert sync.iso_timestamp_to_epoch('2021-11-02 02:01:11.123987+03:00') == 1635807671123
+
+        # Using T as date and time delimiter
+        assert sync.iso_timestamp_to_epoch('2021-11-01T23:01:11') == 1635807671000
+        assert sync.iso_timestamp_to_epoch('2021-11-01T23:01:11.123') == 1635807671123
+        assert sync.iso_timestamp_to_epoch('2021-11-01T23:01:11.123456') == 1635807671123
+        assert sync.iso_timestamp_to_epoch('2021-11-01T23:01:11.123987') == 1635807671123
+        assert sync.iso_timestamp_to_epoch('2021-11-01T23:01:11.123987+00:00') == 1635807671123
+        assert sync.iso_timestamp_to_epoch('2021-11-02T02:01:11.123987+03:00') == 1635807671123
```

Comment:
Is it also expected for all xx:yy values `sync.iso_timestamp_to_epoch('2021-11-02T02:01:11.123987+xx:yy')==1635807671123` ?